src/client/datascience/jupyter/interpreter/jupyterInterpreterSubCommandExecutionService.ts

```diff
@@ -116,6 +120,17 @@ export class JupyterInterpreterSubCommandExecutionService
         // We don't want the process to die when the token is cancelled.
         const spawnOptions = { ...options };
         spawnOptions.token = undefined;
+        const envVars =
+            (await this.activationHelper.getActivatedEnvironmentVariables(undefined, interpreter, true)) || process.env;
+        const jupyterDataPaths = (process.env['JUPYTER_PATH'] || envVars['JUPYTER_PATH'] || '')
+            .split(path.delimiter)
+            .filter((item) => item.trim().length);
+        jupyterDataPaths.push(path.dirname(await this.jupyterPaths.getKernelSpecTempRegistrationFolder()));
+        spawnOptions.env = {
+            ...envVars,
+            JUPYTER_PATH: jupyterDataPaths.join(path.delimiter)
```

Comment:
Magic sauce, this ensures Jupyter finds our kernelspecs (registered in a temporary directory)